main.go

```diff
@@ -16,5 +16,155 @@
 
 package main
 
+import (
+	"context"
+	"net"
+	"net/url"
+	"os"
+	"time"
+
+	nested "github.com/antonfisher/nested-logrus-formatter"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/endpoint"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/authorize"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/kernel"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/ipam/point2pointipam"
+	"github.com/networkservicemesh/sdk/pkg/tools/debug"
+	"github.com/networkservicemesh/sdk/pkg/tools/grpcutils"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/signalctx"
+	"github.com/networkservicemesh/sdk/pkg/tools/spiffejwt"
+)
+
+// Config holds configuration parameters from environment variables
+type Config struct {
+	Name             string        `default:"icmp-server" desc:"Name of ICMP Server"`
+	BaseDir          string        `default:"./" desc:"base directory" split_words:"true"`
+	ListenOn         url.URL       `default:"unix:///listen.on.socket" desc:"url to listen on" split_words:"true"`
+	ConnectTo        url.URL       `default:"unix:///connect.to.socket" desc:"url to connect to" split_words:"true"`
+	MaxTokenLifetime time.Duration `default:"24h" desc:"maximum lifetime of tokens" split_words:"true"`
+
+	EndpointName string `default:"icmp-responder-nse" desc:"url to the local registry to register this NSE"`
+	CidrPrefix   string `default:"169.254.0.0/16" desc:"CIDR Prefix to assign IPs from"`
+}
+
 func main() {
+	// setup context to catch signals
+	ctx := signalctx.WithSignals(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
+
+	// setup logging
```

Comment:
You may wish to consider this style of section comment:
https://github.com/networkservicemesh/cmd-nsc/blob/master/main.go#L75